crates/rpc-client/src/client.rs

```diff
@@ -152,22 +160,33 @@ pub struct RpcClientInner<T> {
     pub(crate) is_local: bool,
     /// The next request ID to use.
     pub(crate) id: AtomicU64,
+    /// The poll interval for the client in milliseconds.
+    pub(crate) poll_interval: AtomicU64,
 }
 
 impl<T> RpcClientInner<T> {
     /// Create a new [`RpcClient`] with the given transport.
+    ///
+    /// Note: Sets the poll interval to 250ms for local transports and 7s for remote transports by
+    /// default.
```

Comment:
we could tweak this based in the chain, because 7s exceeds avg blocktime of a lot of chains.

the alloy_chains::Chain has a function for this.

so we could integrate this in eth_chainId call for example and track whether we've already set the poll_interval based on chain_id